src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -107,6 +110,51 @@ public async Task<IActionResult> GetHostStatus([FromServices] IScriptHostManager
             return Ok(status);
         }
 
+        [HttpGet]
+        [Route("admin/host/workerProcesses")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
+        public async Task<IActionResult> GetWorkerProcesses([FromServices] IScriptHostManager scriptHostManager)
+        {
+            if (!Utility.TryGetHostService(scriptHostManager, out IWebHostRpcWorkerChannelManager webHostLanguageWorkerChannelManager))
+            {
+                return StatusCode(StatusCodes.Status503ServiceUnavailable);
+            }
+
+            var processes = new List<WorkerProcessInfo>();
+            string workerRuntime = _environment.GetFunctionsWorkerRuntime();
+
+            var webhostChannelDictionary = webHostLanguageWorkerChannelManager.GetChannels(workerRuntime);
+
+            List<IRpcWorkerChannel> channels = new List<IRpcWorkerChannel>();
+            if (webhostChannelDictionary is not null)
+            {
+                foreach (var pair in webhostChannelDictionary)
+                {
+                    var workerChannel = await pair.Value.Task;
+                    channels.Add(workerChannel);
+                }
+            }
+
+            if (Utility.TryGetHostService(scriptHostManager, out IJobHostRpcWorkerChannelManager jobHostLanguageWorkerChannelManager))
+            {
+                channels.AddRange(jobHostLanguageWorkerChannelManager.GetChannels(workerRuntime));
+            }
+
+            foreach (var channel in channels)
+            {
+                var processInfo = new WorkerProcessInfo()
```

Comment:
There's already an IHostProcessMonitor (defined in the WebJobs SDK [here](https://github.com/Azure/azure-webjobs-sdk/blob/7bfcd07e520210c9cc092e6c1ac9413fc5c3d178/src/Microsoft.Azure.WebJobs.Host/Scale/DefaultHostProcessMonitor.cs#L15)) that we use for scale status that is all wired up in the Functions Host to monitor both the host and all child processes. Since all this bookeeping is already handled by that component, perhaps it makes sense to just have it also implement a new IHostProcessProvider interface that can be queried for all the process info. That way this is centralized.

Reply:
It would be neat to get the `IHostProcessProvider` implementation working. I think we would not get all the information we need from the proposed `IHostProcessProvider` though, mainly the debug engine info. I am getting some information from the WorkerMetadata object received as part of worker init response. Also, since the current implementation is not introducing any new bookkeeping components, we can easily swap the underlying components without impacting the API if we decide to have `IHostProcessProvider` implementation. Given the sensitive nature of this workitem i would like to merge the current implementation if there are no major red flags.